src/service/resources-impl.js

```diff
@@ -70,6 +70,315 @@ let MarginChangeDef;
+   * may request the Resources manager to update children's inViewport state.
+   * A child's inViewport state is a logical AND between inLocalViewport
+   * specified here and parent's own inViewport state.
+   * @param {!Element} parentElement
+   * @param {!Element|!Array<!Element>} subElements
+   * @param {boolean} inLocalViewport
+   */
+  updateInViewport(parentElement, subElements, inLocalViewport) {}
+}
+
+/**
+ * @interface
+ */
+class MutatorsAndOwnersDef extends OwnersDef {
+  /**
+   * Requests the runtime to change the element's size. When the size is
+   * successfully updated then the opt_callback is called.
+   * @param {!Element} element
+   * @param {number|undefined} newHeight
+   * @param {number|undefined} newWidth
+   * @param {function()=} opt_callback A callback function.
+   * @param {!../layout-rect.LayoutMarginsChangeDef=} opt_newMargins
+   */
+  changeSize(element, newHeight, newWidth, opt_callback, opt_newMargins) {}
+
+  /**
+   * Return a promise that requests the runtime to update the size of
+   * this element to the specified value.
+   * The runtime will schedule this request and attempt to process it
+   * as soon as possible. However, unlike in {@link changeSize}, the runtime
+   * may refuse to make a change in which case it will reject promise, call the
+   * `overflowCallback` method on the target resource with the height value.
+   * Overflow callback is expected to provide the reader with the user action
+   * to update the height manually.
+   * Note that the runtime does not call the `overflowCallback` method if the
+   * requested height is 0 or negative.
+   * If the height is successfully updated then the promise is resolved.
+   * @param {!Element} element
+   * @param {number|undefined} newHeight
+   * @param {number|undefined} newWidth
+   * @param {!../layout-rect.LayoutMarginsChangeDef=} opt_newMargins
+   * @return {!Promise}
+   */
+  attemptChangeSize(element, newHeight, newWidth, opt_newMargins) {}
+
+  /**
+   * Expands the element.
+   * @param {!Element} element
+   */
+  expandElement(element) {}
+
+  /**
+   * Return a promise that requests runtime to collapse this element.
+   * The runtime will schedule this request and first attempt to resize
+   * the element to height and width 0. If success runtime will set element
+   * display to none, and notify element owner of this collapse.
+   * @param {!Element} element
+   * @return {!Promise}
+   */
+  attemptCollapse(element) {}
+
+  /**
+   * Collapses the element: ensures that it's `display:none`, notifies its
+   * owner and updates the layout box.
+   * @param {!Element} element
+   */
+  collapseElement(element) {}
+
+  /**
+   * Runs the specified measure, which is called in the "measure" vsync phase.
+   * This is simply a proxy to the privileged vsync service.
+   *
+   * @param {function()} measurer
+   * @return {!Promise}
+   */
+  measureElement(measurer) {}
```

Comment:
I don't think these belong as an Owners interface, they're meant to be used by elements themselves.

Reply:
This is not part of Owners interface, it's part of Mutators interface